### PR TITLE
[DirectX][NFC] Rename ShaderFlag to SHADER_FEATURE_FLAG.

### PR DESCRIPTION
**Pull request by python3kgae**
This is preparation for add ShaderFlag in DXIL.

For #57925

### CODE REVIEW

File: llvm/lib/ObjectYAML/DXContainerYAML.cpp [hunk 1]
```diff
@@ -24,14 +24,14 @@ static_assert((uint64_t)dxbc::FeatureFlags::NextUnusedBit <= 1ull << 63,
               "Shader flag bits exceed enum size.");
 
 DXContainerYAML::ShaderFlags::ShaderFlags(uint64_t FlagData) {
-#define SHADER_FLAG(Num, Val, Str)                                             \
+#define SHADER_FEATURE_INFO(Num, Val, Str)                                     \
```
Comment (llvm-beanz) [inline llvm/lib/ObjectYAML/DXContainerYAML.cpp:27]: I think this renaming makes this more confusing not less. We probably have a problem that we have too many things called "shader flags", but only partially changing the name makes this more confusing.

A name like "shader feature flags" might be a better name and could be differentiated from the "shader module flags" that are present in the DXIL metadata, but if we're going to do a renaming it should be comprehensive and not leave us in a confusing state.

Comment (python3kgae) [reply llvm/lib/ObjectYAML/DXContainerYAML.cpp]: Updated to SHADER_FEATURE_FLAG

File: llvm/include/llvm/ObjectYAML/DXContainerYAML.h [hunk 2]
```diff
@@ -56,7 +56,7 @@ struct DXILProgram {
   std::optional<std::vector<llvm::yaml::Hex8>> DXIL;
 };
 
-#define SHADER_FLAG(Num, Val, Str) bool Val = false;
+#define SHADER_FEATURE_FLAG(Num, Val, Str) bool Val = false;
 struct ShaderFlags {
```
Comment (llvm-beanz) [inline llvm/include/llvm/ObjectYAML/DXContainerYAML.h:60]: Should this also be `FeatureFlags`?

Comment (python3kgae) [reply llvm/include/llvm/ObjectYAML/DXContainerYAML.h:60]: There're 39 DxilModelFlags and 33 FeatureFlags in dxc currently.
Maybe we should save DxilModelFlags and generate FeatureFlags from DxilModelFlags?

Comment (python3kgae) [reply llvm/include/llvm/ObjectYAML/DXContainerYAML.h:60]: Updated to ShaderFeatureFlags.

**Review by llvm-beanz — APPROVED** [review]

**Review by bob80905 — COMMENTED** [review]
Can you change the title of the PR so that it's SHADER_FEATURE_FLAG?

**Review by bob80905 — APPROVED** [review]

**Comment by python3kgae** [comment]
> Can you change the title of the PR so that it's SHADER_FEATURE_FLAG?

Updated.

**Pull request merged**